components/style/values/computed/counters.rs

```diff
@@ -4,11 +4,202 @@
+            return Ok(Content::None)
+        }
+        if input.try(|input| input.expect_ident_matching("-moz-alt-content")).is_ok() {
+            #[cfg(feature = "gecko")]
+            return Ok(Content::MozAltContent)
+        }
+
+        let mut content = vec![];
+        loop {
+            match input.try(|i| SpecifiedUrl::parse(context, i)) {
+                #[cfg(feature = "gecko")]
+                Ok(mut url) => {
+                    url.build_image_value();
+                    content.push(ContentItem::Url(url));
+                    continue;
+                }
+                _ => (),
+            }
+            // FIXME: remove clone() when lifetimes are non-lexical
+            match input.next().map(|t| t.clone()) {
+                Ok(Token::QuotedString(ref value)) => {
+                    content.push(ContentItem::String(value.as_ref().to_owned()))
+                }
+                Ok(Token::Function(ref name)) => {
+                    let result = match_ignore_ascii_case! { &name,
+                        "counter" => Some(input.parse_nested_block(|input| {
+                            let name = input.expect_ident()?.as_ref().to_owned();
+                            #[cfg(feature = "servo")]
+                            let style = Content::parse_counter_style(input);
+                            #[cfg(feature = "gecko")]
+                            let style = Content::parse_counter_style(context, input);
+                            Ok(ContentItem::Counter(name, style))
+                        })),
+                        "counters" => Some(input.parse_nested_block(|input| {
+                            let name = input.expect_ident()?.as_ref().to_owned();
+                            input.expect_comma()?;
+                            let separator = input.expect_string()?.as_ref().to_owned();
+                            #[cfg(feature = "servo")]
+                            let style = Content::parse_counter_style(input);
+                            #[cfg(feature = "gecko")]
+                            let style = Content::parse_counter_style(context, input);
+                            Ok(ContentItem::Counters(name, separator, style))
+                        })),
+                        #[cfg(feature = "gecko")]
+                        "attr" => Some(input.parse_nested_block(|input| {
+                            Ok(ContentItem::Attr(Attr::parse_function(context, input)?))
+                        })),
+                        _ => None
+                    };
+                    match result {
+                        Some(result) => content.push(result?),
+                        None => return Err(input.new_custom_error(
+                            StyleParseErrorKind::UnexpectedFunction(name.clone())
+                        ))
+                    }
+                }
+                Ok(Token::Ident(ref ident)) => {
+                    let valid = match_ignore_ascii_case! { &ident,
```

Comment:
While we're here we can just clean this up:

content.push(match_ignore_ascii_case! { &ident,
    "open-quote" => ContentItem::OpenQuote,
    // ...
    _ => return Err(..),
})